src/zarr/core/buffer/core.py

```diff
@@ -470,7 +470,9 @@ def all_equal(self, other: Any, equal_nan: bool = True) -> bool:
         # every single time we have to write data?
         _data, other = np.broadcast_arrays(self._data, other)
         return np.array_equal(
-            self._data, other, equal_nan=equal_nan if self._data.dtype.kind not in "USTO" else False
+            self._data,
+            other,
+            equal_nan=equal_nan if self._data.dtype.kind not in "USTOV" else False,
```

Comment:
can we factor this check out into a stand-alone boolean function?

Reply:
Where? Is there somewhere this would be re-used in the core?

Reply:
The change here is just to add "V", so I wouldn't change it now